azure/converters/inboundnatrules.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package converters
+
+import (
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-02-01/network"
+)
+
+type ExistingInboundNatSpec struct {
```

Comment:
Looks like the only place we are using this is converting it to a map of ports in use https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1870/files#diff-64818b5b4ad20088a23cd553a259558754c7f3a3b73813d166b307c0f44a694cR64-R67

Have you considered getting rid of this intermediate struct and directly storing usedPorts or portsInUsed as a `map[int32]struct{}` ?